### DISCUSSION TITLE
pagination with graphql api

### DISCUSSION THREAD

**Discussion by rpimentaf** [komarovalexander/ka-table#210] [Q&A] [answered]
Hi, @komarovalexander!
I have a graphql api with pagination that returns data in pieces (pages). In that situation, the ka-table pagination does not work properly because although the total records be over than 1 page, the ka-table rendered only shows 1 page icon.
Do you know any suggestion to configure tableprops.paging.pageIndex based on the total records to be returned by the api and it not be reconfigured?
Tks in advance!!!

**Comment by komarovalexander** [comment, marked as answer]
HI @rpimentaf 
please have a look on https://komarovalexander.github.io/ka-table/#/remote-data-editing demo
it has 
```
   dispatch(updatePagesCount(result.pagesCount));
```
looks like this is what you need

----
pagesCount sets the count of pages for specific data and disables local data pagination:
```
paging: {
 pagesCount?: number; 
 // ....
}
```
 the action creator to update pagesCount value:
```
updatePagesCount(pagesCount: number)
```

**Comment by rpimentaf** [comment]
@komarovalexander , tks very much! Exactly what i needed.
